java/src/main/native/CMakeLists.txt

```diff
@@ -15,9 +15,35 @@
 #=============================================================================
 cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
 
+# Use GPU_ARCHS if CMAKE_CUDA_ARCHITECTURES is not defined
+if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES AND DEFINED GPU_ARCHS)
```

Comment:
:clap: I wish this were there in `cpp/CMakeLists.txt` as well.

Reply:
I felt I had to do it this way since I couldn't find a clean way to specify undefined vs defined-as-empty variables in the Maven pom, so I left the existing GPU_ARCHS behavior as-is and translated it here.